gocourse07/birdaviary/internal/storage/storage.go

```diff
@@ -0,0 +1,31 @@
+package storage
+
+import (
+	"sync"
+
+	"birdaviary/internal/logger"
+)
+
+type Storage struct {
+	data map[string][]int
+	mu   sync.Mutex
+	log  *logger.Logger
+}
+
+func NewStorage(log *logger.Logger) *Storage {
+	return &Storage{
+		data: make(map[string][]int),
+		log:  log,
+	}
+}
+
+func (m *Storage) Save(sensorName string, value int) {
+	defer m.mu.Unlock()
+	m.mu.Lock()
+	m.data[sensorName] = append(m.data[sensorName], value)
```

Comment:
А хіба потрібно захищати мютексом `m.log`?
```suggestion
	m.mu.Lock()
	m.data[sensorName] = append(m.data[sensorName], value)
	m.mu.Unlock()
```